### ISSUE TITLE
Issue of Waymo labels I/O

### ISSUE THREAD

**Issue by AronCao49** [dvlab-research/SphereFormer#24]
First of all, thanks for your effort and your great work! Currently, I have been trying to train SphereFormer on the Waymo dataset. However, after converting all data to the KITTI format, I found there is an issue when loading the converted labels of Waymo. 

Specifically, the raw semantic labels of Waymo seem to be of shape [N, 2], where [N, 0] is the instance label for panoptic segmentation and [N, 1] is the label indicating the semantic class, shown as follows:

https://github.com/dvlab-research/SphereFormer/blob/63af0fa93e6510923f6a6812fce990151a238517/util/waymo.py#L111-L112

The current dataloader resizes the raw labels in (-1, 1) which results in a label of [2N, 1], causing an error when attempting to train with Waymo. To fix this error, one simply changes the command of line 111 into:
```
annotated_data = np.fromfile(file_path.replace('velodyne', 'labels')[:-3] + 'label', 
                                 dtype=np.uint32).reshape((-1, 2)) 
annocated_data = annocated_data[:, 1]
```
Then the Waymo training code runs smoothly.

**Comment by X-Lai** [comment]
Hi, thanks for your interest in our work. Did you use our released data processing script to convert the original waymo raw data to the SemanticKITTI format?

**Comment by AronCao49** [comment]
Hi, thanks for your prompt reply. I did convert the waymo data to KITTI format following the official instruction, but that led to the error I mentioned above. 

According to the current code, the semantic labels extracted seem to contain both the label for the semantic class and the instance ID (with a shape of [N, 2]):
https://github.com/dvlab-research/SphereFormer/blob/63af0fa93e6510923f6a6812fce990151a238517/data/waymo_to_semanticKITTI/lib/utils.py#L20-L37

The current waymo dataloader seems to directly load these [N, 2] labels as training targets without removing the redundant instance ID part, which would trigger a CUDA side error during the training process. 

At least in my case, using the waymo dataset of v1.3.0, I need to manually include the modification I mentioned before to avoid this error. May I know if you can successfully conduct the waymo training with the current code?

**Comment by X-Lai** [comment]
Thanks for pointing out this issue. I may miss this problem when refactoring this codebase.

**Comment by normster** [comment]
I spent several hours trying to debug this before thinking to check the issues. Thank you @AronCao49 for documenting the fix!

**Comment by X-Lai** [comment]
Thanks for reporting this issue. I just fixed this bug and committed to the master branch.

**Comment by rockywind** [comment]
Hi, @AronCao49 @X-Lai @normster 
I use the latest released data processing script to convert the original waymo raw data to the SemanticKITTI format and use the semantic kitti label color to visualize the converted to semantic kitti format point cloud.
I found the semantic visualization is bad. I found the label of point cloud is unconsistency!
![image](https://github.com/dvlab-research/SphereFormer/assets/14838781/ce517c39-efa7-4faa-a750-ff379c815f7a)
